hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/ProfilingFileIoEvents.java

```diff
@@ -80,7 +80,7 @@ public void afterMetadataOp(@Nullable FsVolumeSpi volume,
 
   public long beforeFileIo(@Nullable FsVolumeSpi volume,
       FileIoProvider.OPERATION op, long len) {
-    if (isEnabled && ThreadLocalRandom.current().nextInt() < sampleRangeMax) {
+    if (isEnabled && ThreadLocalRandom.current().nextInt(Integer.MAX_VALUE) < sampleRangeMax) {
```

Comment:
Great catch! some other same logic should be corrected too, such as org.apache.hadoop.hdfs.client.impl.metrics.BlockReaderIoProvider#read, please also check them together.